api/repo.go

```diff
@@ -650,6 +670,35 @@ func UpdateRepo(c *gin.Context) {
 		r.SetBranch(input.GetBranch())
 	}
 
+	// update build limit if set
+	if input.GetBuildLimit() > 0 && maxBuildLimit == 0 {
+		// allow build limit between 1 - 30
+		r.SetBuildLimit(
+			int64(
+				util.MaxInt(
+					constants.BuildLimitMin,
+					util.MinInt(
+						int(input.GetBuildLimit()),
+						constants.BuildLimitMax,
+					), // clamp max
+				), // clamp min
+			),
+		)
+	} else if input.GetBuildLimit() > 0 {
+		// allow build limit between 1 - value configured by server
+		r.SetBuildLimit(
+			int64(
+				util.MaxInt(
+					constants.BuildLimitMin,
+					util.MinInt(
+						int(input.GetBuildLimit()),
+						int(maxBuildLimit),
+					), // clamp max
+				), // clamp min
+			),
+		)
+	}
```

Comment:
This seems redundant since `maxBuildLimit` defaults to `constants.BuildLimitMax`

Reply:
Yeah I can definitely see where you're coming from on that 😅 

In most cases, the code above will be used since `maxBuildLimit` **should** contain a value above `0`.

However, it's possible for someone to accidentally configure `maxBuildLimit` with a value of `0`.

In this scenario, the above code wouldn't executed and the following would run instead:

https://github.com/go-vela/server/blob/3c1b13d8471a7f460d6cd6ecba259b8d18df7668/api/repo.go#L674-L687

So, you could view the redundant code as providing an implicit fallback for the build limit on the server?

The only reason I can come up with to keep it has to do with misconfiguration on the server.

If we're thinking we don't want the extra code to handle that scenario, then I can remove it?

Reply:
Feels like we should add some validation logic for the `maxBuildLimit` value to account for that scenario.

Reply:
Sure 👍  I'll add that validation logic now.

FWIW:

We may have to remove this validation logic in the future if the use case comes up.

It's doubtful, but at some point a user running Vela may want to set no max build limit for repos.

A pattern I've seen in the Go community (and outside of it) is `0` == unlimited when setting values like this.

Reply:
This should be addressed 👍